wagn/rails/controllers/card_controller.rb

```diff
@@ -73,6 +73,11 @@ def per_request_setup
     Card::Cache.renew
     Card::Env.reset :controller=>self
     Card::Auth.set_current_from_session
+    
+    if params[:id] && !params[:id].valid_encoding?  # slightly better way to handle encoding issues (than the rescue in load_id)
+                                                    # we should find the place where we produce these bad urls
```

Comment:
Unfortunately airbrake can't tell us where link is coming from. The last errors are all produced by +link_and_compare addresses, for example http://wikirate.org/Amalgamated_Plantations_Pvt_Ltd_+Human_Rights+L_Or%e9al+link_and_compare. The remote ip belongs Hetzner Online, that's Wikirate's server host, right?